clang-cl compiler not found on Windows

**Describe the bug**
`No CMAKE_CXX_COMPILER could be found.` when trying to build with clang-cl with cmake on Windows. This just happens with the latest vcpkg version, previous ones worked.

**Environment**
- OS: Windows
- Compiler: clang-cl latest

**To Reproduce**
```
git clone https://github.com/jsmolka/NanoboyAdvance
cd NanoboyAdvance
mkdir build
cd build
set VCPKG_ROOT=path/to/vcpkg
cmake -T clangcl ..
```

Full build instructions in the [readme](https://github.com/fleroviux/NanoboyAdvance#windows).

**Expected behavior**
The previous vcpkg version managed to compile.

**Failure logs** 
```
-- Building for: Visual Studio 16 2019
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.18363.
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:8 (project):
  No CMAKE_CXX_COMPILER could be found.



-- Configuring incomplete, errors occurred!
See also "C:/dev/code/NanoboyAdvance/build/CMakeFiles/CMakeOutput.log".
See also "C:/dev/code/NanoboyAdvance/build/CMakeFiles/CMakeError.log".
```

Logs are attached (some areas in German, I can translate if necessary).
[CMakeError.log](https://github.com/microsoft/vcpkg/files/4880961/CMakeError.log)
[CMakeOutput.log](https://github.com/microsoft/vcpkg/files/4880962/CMakeOutput.log)

**Additional context**
I first noticed this issue in a GitHub Actions workflow and therefore created an issue in the actions repo. The [answer](https://github.com/actions/virtual-environments/issues/1152#issuecomment-654042618) is worth a read and gives a possible explanation.

> selects Visual Studio 16 2019 as default generator

`14.26.28801\lib\x64\libcmt.lib : warning LNK4272: library machine type 'x64' conflicts with target machine type 'x86'`

`Visual Studio 16 2019 -A x64` << -- x64 default generator 
?

"C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/Llvm/bin/clang-cl.exe" --help

x64 Llvm -->> "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/Llvm/x64/"

--- a/scripts/buildsystems/msbuild/vcpkg-general.xml
+++ b/scripts/buildsystems/msbuild/vcpkg-general.xml
@@ -37,7 +37,7 @@
   </BoolProperty>
   <BoolProperty Name="VcpkgAutoLink"
                 DisplayName="Use AutoLink"
-                Description="Enables automatic linking with libraries build using Vcpkg"
+                Description="Enables automatic linking with libraries build using Vcpkg. Does not work with lld-link.exe."
                 Category="General"
                 Default="true">
   </BoolProperty>

--- a/scripts/buildsystems/msbuild/vcpkg.targets
+++ b/scripts/buildsystems/msbuild/vcpkg.targets
@@ -64,7 +64,7 @@
     <VcpkgApplocalDeps Condition="'$(VcpkgApplocalDeps)' == ''">true</VcpkgApplocalDeps>
     <!-- Deactivate Autolinking if lld is used as a linker. (Until a better way to solve the problem is found!).
     Tried to add /lib as a parameter to the linker call but was unable to find a way to pass it as the first parameter. -->
-    <VcpkgAutoLink Condition="'$(UseLldLink)' == 'true' and '$(VcpkgAutoLink)' == ''">false</VcpkgAutoLink>
+    <VcpkgAutoLink Condition="'$(UseLldLink)' == 'true'">false</VcpkgAutoLink>
   </PropertyGroup>
 
   <ItemDefinitionGroup Condition="'$(VcpkgEnabled)' == 'true'">

